source/OptimaJet.DWKit.StarterApplication/Services/Workflow/WorkflowProjectService.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Hangfire.Server;
+using Microsoft.EntityFrameworkCore;
+using OptimaJet.DWKit.Application;
+using OptimaJet.DWKit.Core;
+using OptimaJet.DWKit.Core.Model;
+using OptimaJet.DWKit.StarterApplication.Models;
+using OptimaJet.DWKit.StarterApplication.Repositories;
+using OptimaJet.Workflow.Core.Runtime;
+using Serilog;
+
+namespace OptimaJet.DWKit.StarterApplication.Services.Workflow
+{
+    public class WorkflowProjectService
+    {
+
+        public IJobRepository<Project> ProjectRepository { get; }
+        public IJobRepository<UserTask> TaskRepository { get; }
+        public IJobRepository<User> UserRepository { get; }
+
+        public WorkflowProjectService(
+            IJobRepository<Project> projectRepository,
+            IJobRepository<UserTask> taskRepository,
+            IJobRepository<User> userRepository
+        )
+        {
+            ProjectRepository = projectRepository;
+            TaskRepository = taskRepository;
+            UserRepository = userRepository;
+        }
+
+        public async Task ReassignUserTasks(int projectId, int previousOwnerId, int newOwnerId) {
+          var project = await this.ProjectRepository.Get()
+              .Include(p => p.Products)
+              .ThenInclude(product => product.UserTasks)
+              .Where(p => p.Id == projectId)
+              .FirstOrDefaultAsync();
+          var previousOwner = await this.UserRepository.GetAsync(previousOwnerId);
+          var newOwner = await this.UserRepository.GetAsync(newOwnerId);
+
+          project.Products.ForEach(product => {
+              this.ReassignUserTasksForProduct(product, previousOwner, newOwner);
+          });
+        }
+
+        private void ReassignUserTasksForProduct(Product product, User previous, User next) {
+          product.UserTasks
```

Comment:
@chrisvire what else has to change?

I was looking at the dashboard at http://localhost:7081/
and the task in there appropriately updates

Also, I suppose this type of reassignment is a bit naiive, as it assumes both users have the same permissions.
It's totally possible that the previous user is an org admin, and would retain some tasks, just because they are an org admin, yeah?

Reply:
Would it be better to, using code from the WorkflowProductService, to remove and then re-create? rather than re-assign?

```cs
// parts of 
ProductProcessChangedAsync(...)
// may be relevant
```

Reply:
yeah actually, it looks like if I can create a `ProductProcessChangedArgs`, I should just be able to reuse that method entirely